_posts/2021-05-21-js-waku.md

```diff
@@ -0,0 +1,171 @@
+
+Key Waku v2 protocols are already available:
+[message](https://rfc.vac.dev/spec/14/), [store](https://rfc.vac.dev/spec/13/), [relay](https://rfc.vac.dev/spec/11/) and [light push](https://rfc.vac.dev/spec/19/),
+enabling your dApp to:
+
+- Send and receive near-instant messages on the Waku network (relay),
+- Query nodes for messages that may have been missed, e.g. due to poor cellular network (store),
+- Send messages with confirmations (light push).
+
+JS-Waku needs to operate in the same context from which Waku v2 was born:
+a restricted environment were connectivity or uptime are not guaranteed;
+JS-Waku bring Waku v2 to the browser.
+
+## Achievements so far
+
+We focused the past month on developing a [ReactJS Chat App](https://status-im.github.io/js-waku/).
+The aim was to create enough building blocks in JS-Waku to enable this showcase web app that
+we now [use for dogfooding](https://github.com/status-im/nim-waku/issues/399) purposes.
+
+Most of the effort was on getting familiar with the [js-libp2p](https://github.com/libp2p/js-libp2p) library
+that we heavily rely on.
+JS-Waku is the second implementation of Waku v2 protocol,
+so a lot of effort on interoperability was needed.
+For example, to ensure compatibility with the nim-waku reference implementation,
+we run our [tests against wakunode2](https://github.com/status-im/js-waku/blob/90c90dea11dfd1277f530cf5d683fb92992fe141/src/lib/waku_relay/index.spec.ts#L137) as part of the CI.
+
+This interoperability effort helped solidify the current Waku v2 specifications:
+By clarifying the usage of topics
+([#327](https://github.com/vacp2p/rfc/issues/327), [#383](https://github.com/vacp2p/rfc/pull/383)),
+fix discrepancies between specs and nim-waku
+([#418](https://github.com/status-im/nim-waku/issues/418), [#419](https://github.com/status-im/nim-waku/issues/419))
+and fix small nim-waku & nim-libp2p bugs
+([#411](https://github.com/status-im/nim-waku/issues/411), [#439](https://github.com/status-im/nim-waku/issues/439)).
+
+To fully access the waku network, JS-Waku needs to enable web apps to connect to nim-waku nodes.
+A standard way to do so is using secure websockets as it is not possible to connect directly to a TCP port from the browser.
+Unfortunately websocket support is not yet available in [nim-libp2p](https://github.com/status-im/nim-libp2p/issues/407) so 
+we ended up deploying [websockify](https://github.com/novnc/websockify) alongside wakunode2 instances.
+
+As we built the [web chat app](https://github.com/status-im/js-waku/tree/main/examples/web-chat),
+we were able to fine tune the API to provide a simple and succinct interface.
+You can start a node, connect to other nodes and send a message in less than ten lines of code:
+
+```javascript
+import { Waku } from 'js-waku';
+
+const waku = await Waku.create({});
+
+const nodes = await getStatusFleetNodes();
```

Comment:
side note but I don't think we want to hardcode Status functions stuff here, this should be a static file that you load where it being hosted by Status is an implementation detail.

Reply:
Would the static file be in the js-waku repo? what would it contain?
I expect to have a more generic discovery method once we progress in this area. I believe for the sake of transparency it was best to clarify in the API name that we connect to Status nodes.